EasyClangComplete.py

```diff
@@ -169,7 +169,22 @@ def on_close(self, view):
             log.debug(" closing view %s", view.buffer_id())
             if not self.completer:
                 return
-            self.completer.remove(view.buffer_id())
+            future = EasyClangComplete.pool_read.submit(
+                self.completer.remove, view.buffer_id())
+            future.add_done_callback(EasyClangComplete.completer_removed)
```

Comment:
Ideally you would also cancel active threads for given ID but that would require storing managing active futures.